File: code/data_structures/src/tree/binary_tree/binary_tree/traversal/postorder/postorder.cpp
#include <iostream>
#include <memory>
// Part of Cosmos by OpenGenus Foundation
struct node
{
    int data;
    node *left;
    node *right;
};

node *newnode(int data)
{
     node *temp = new node;
     temp->data = data;
     temp->left = nullptr;
     temp->right = nullptr;

     return temp;
}

int post_order(node *root)
{
    if (!root)
        return 0;

    post_order(root->left);
    post_order(root->right);
    std::cout << root->data << " ";
}

File: code/data_structures/src/tree/binary_tree/binary_tree/traversal/postorder/postorder.py
# Part of Cosmos by OpenGenus Foundation
class node:
    def __init__(self, value):
        self.data = value
        self.left = None
        self.right = None


def post_order(root):
    if root is not None:
        post_order(root.left)
        post_order(root.right)
        print(root.data, end=" ")


if __name__ == '__main__':
    root = node(1)
    root.left = node(2)
    root.right = node(3)
    root.left.left = node(4)
    root.left.right = node(5)
    root.right.left = node(6)
    root.right.right = node(7)

    print("Post order traversal of tree is:", end=" ")
    post_order(root)
